tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2251,50 +2245,61 @@ private[optimizer] abstract class OptimizerCore(
                  * will not go into infinite recursion.
                  */
                 implicit val scope = Scope.Empty
-                inline(Nil, Some(PreTransTree(x)), Nil, target, isStat = false,
+                inline(Nil, Some(x), Nil, target, isStat = false,
                     usePreTransform = false)(finishTransform(isStat = false))
-              }
+              })
```

Comment:
Because they never make optimistic choices (`tryOrRollback`) nor call any method that can make ones (except this weird pair of hacks, but well ...). Therefore, then don't need to. And since they don't need to, they'd better not do it, as it's expensive.